Add documentation for Traces beta feature (#7004)
dagger/dagger: docs/current_docs/manuals/user/cloud/572923-get-started.mdx

    :::
  - Add the secret to your CI environment as a variable named `DAGGER_CLOUD_TOKEN`.
- For **Team** plan subscribers with ephemeral CI runners only: Add a wait step (example below) for the Docker configuration to allow the Dagger Engine to push cache volumes to the experimental Dagger Cloud distributed cache. Subscribers on the **Individual** plan do not have access to the experimental distributed cache and do not need to add this step.

> - If you are using GitHub Actions, install the Dagger Cloud GitHub app for GitHub Checks. This app adds a GitHub Check to your pull requests for your Dagger pipeline run and links to the pipeline visualization for the commit in Dagger Cloud.

Reviewer: I found this ambiguous so here is an attempt to make it clearer. I may have got it wrong though.

```suggestion
- If you are using GitHub Actions, install the Dagger Cloud GitHub app for GitHub Checks. This app adds a GitHub check to your pull requests. Each check links to the corresponding pipeline visualization in Dagger Cloud.
```

Author: The app has a new UX, which prints the command out. For `dagger call`, you get the status of the run reported back to the check, which is a big improvement. I'm waiting for clarification from @aluzzardi about whether this is a NEW app or an update to the existing one.

### Trigger runs from your local development host

Before completing these steps, ensure you've [added your Dagger Cloud token as an environment variable](#connect-from-your-local-development-host).

1. In your terminal, call the Dagger Module. If your environment variable is properly defined, you will see a Dagger Cloud URL printed in the terminal.
1. Click on the URL to view details of the run in Dagger Cloud.

:::info

> Dagger 0.11.0 adds support for visualizing Dagger functions but does not yet include a way to print the Dagger Cloud URL in the CLI output. We're adding this soon.

Reviewer: ```suggestion
Dagger 0.11.0 adds support for visualizing Dagger functions but does not yet print the Dagger Cloud URL for the pipeline run in the CLI output. We're adding this soon.
```

git push
```

> ### Visualizing Dagger pipelines

Reviewer: ```suggestion
### Visualize Dagger pipelines
```

...for consistency with other headings,which start with a verb

:::info

> Dagger 0.11.0 adds support for visualizing Dagger Functions calls in Dagger Cloud. With this update, Dagger Cloud will default to the use the new Traces experience for both `dagger call` and `dagger run` commands. Engines running Dagger 0.11.0 or later will automatically use the new Traces experience. Engines running Dagger 0.10.0 or earlier will continue to use the legacy Runs experience.

Reviewer: ```suggestion
Dagger 0.11.0 adds support for visualizing Dagger Function calls in Dagger Cloud. With this update, Dagger Cloud will default to use the new Traces experience for both `dagger call` and `dagger run` commands. Dagger v0.11.0 or later will automatically use the new Traces experience. Older versions of Dagger will continue to use the legacy Runs experience.
```

Reviewer: I am updating to use just `Dagger` here instead of `Engines running Dagger` because afaik in this context, it is the same thing.

> Dagger 0.11.0 adds support for visualizing Dagger Functions calls in Dagger Cloud. With this update, Dagger Cloud will default to the use the new Traces experience for both `dagger call` and `dagger run` commands. Engines running Dagger 0.11.0 or later will automatically use the new Traces experience. Engines running Dagger 0.10.0 or earlier will continue to use the legacy Runs experience.

Reviewer: Question, is it `Traces` or `traces`? I see both styles used, should we fix on one?

Author: Capital T

:::

#### Traces
Navigate to the **Traces** page of the Dagger Cloud dashboard. You should see your most recent Dagger pipeline run as the first entry in the list, as shown below: 

![View Traces](/img/current_docs/manuals/user/cloud/get-started/view-traces.png)

A Trace represents one invocation of a Dagger pipeline, run locally or in CI. It contains detailed information about the steps performed by the pipeline.

The **Traces** page provides an overview of all Dagger pipeline runs. You can drill down into the details of a specific run by clicking it. This directs you to a **Trace Details** page, as shown below:

![View Trace Details](/img/current_docs/manuals/user/cloud/get-started/view-trace-details.png)

> The **Trace Details** page includes detailed status and duration metadata about the pipeline steps. If there are any errors in the run, Dagger Cloud automatically brings you to the first error in the list. You see detailed logs and output of each step, making it easy for you to debug your pipelines and collaborate with your teammates.

Reviewer 2: > If there are any errors in the run, Dagger Cloud automatically brings you to the first error in the list.

I'm not sure that's the case yet. cc @vito

 A Dagger trace shows a detailed timeline for operations completed in a pipeline run. Dagger Cloud also applies unique IDs and icons for related operations, which are essential for debugging your pipeline when something does wrong. To learn more about the Traces view and how to debug a failure, see the [Dagger Cloud user interface documentation](./741031-user-interface.mdx).

#### Dagger Runs

> :::info

Reviewer: ```suggestion
:::warning
```

> The Runs view only applies to Dagger pipeline runs done with engines before version 0.11.0. 

Reviewer: ```suggestion
The Runs view only applies to pipeline runs with Dagger versions older than v0.11.0. 
```
